improver_tests/acceptance/acceptance.py

```diff
@@ -84,6 +89,129 @@ def cli_name_with_dashes(dunder_file):
     return module_dashes
 
 
+@functools.lru_cache()
+def acceptance_checksums(checksum_path=None):
+    """
+    Retrieve a list of checksums from file in text list format, as produced by
+    the sha256sum command line tool.
+
+    Args:
+        checksum_path (Optional[pathlib.Path]): Path to checksum file. File
+            should be plain text in the format produced by the sha256sum
+            command line tool. Paths listed in the file should be relative to
+            the KGO root directory found by kgo_root().
+
+    Returns:
+        Dict[pathlib.Path, str]: Dict with keys being relative paths and
+            values being hexadecimal checksums
+    """
+    if checksum_path is None:
+        checksum_path = DEFAULT_CHECKSUM_FILE
+    if not kgo_exists():
+        raise IOError("KGO path must be defined")
+    with open(checksum_path, mode="r") as checksum_file:
+        checksum_lines = checksum_file.readlines()
+    checksums = {}
+    for line in checksum_lines:
+        parts = line.strip().split("  ", maxsplit=1)
+        csum = parts[0]
+        path = pathlib.Path(parts[1])
+        checksums[path] = csum
+    return checksums
+
+
+def verify_checksum(kgo_path, checksums=None, checksum_path=None):
+    """
+    Verify an individual KGO file's checksum.
+
+    Args:
+        kgo_path (pathlib.Path): Path to file in KGO directory
+        checksums (Optional[Dict[pathlib.Path, str]]): Lookup dictionary
+            mapping from paths to hexadecimal checksums. If provided, used in
+            preference to checksum_path.
+        checksum_path (pathlib.Path): Path to checksum file. File should be
+            plain text in the format produced by the sha256sum command line
+            tool.
+
+    Raises:
+        KeyError: File being verified is not found in checksum dict/file
+        ValueError: Checksum does not match value in checksum dict/file
+    """
+    if checksums is None:
+        checksums = acceptance_checksums(checksum_path)
+    kgo_csum = calculate_checksum(kgo_path)
+    kgo_norm_path = pathlib.Path(os.path.normpath(kgo_path))
+    kgo_rel_path = kgo_norm_path.relative_to(kgo_root())
+    try:
+        expected_csum = checksums[kgo_rel_path]
+    except KeyError:
+        msg = f"Checksum for {kgo_rel_path} is missing"
```

Comment:
It'll take some rearrangement to provide information about where the checksums come from, but it would be a better explanation of what to fix for a developer who sees this exception message. I'll have a go at changing this.

The length check in `test_kgo_checksums` will pick up removed files, since it compares the number of files found by crawling the directory against the number of lines in the checksum file.